synapse/tests/test_utils_getrefs.py

```diff
@@ -45,30 +38,33 @@ def test_basics(self):
         with self.raises(s_exc.BadArg):
             s_getrefs.download_refs_handler('http://raw.githubusercontent.com/../../attack-flow/attack-flow-schema-2.0.0.json')
 
-        filename = pathlib.Path(s_data.path(
-            'jsonschemas',
-            'raw.githubusercontent.com',
-            'oasis-open',
-            'cti-stix2-json-schemas',
-            'stix2.1',
-            'schemas',
-            'common',
-            'core.json'
-        ))
-
-        self.true(filename.exists())
-        filename.unlink()
-
-        # Clear the cached validator funcs so the ref handlers in s_getrefs get called
-        s_config._JsValidators = {}
-
-        with self.getLoggerStream('synapse.utils.getrefs') as stream:
-            with self.getVcr() as cass:
-                s_getrefs.main(args)
-
-        stream.seek(0)
-        mesgs = stream.read()
-        mesg = f'Downloading schema from {CORE_URL}.'
-        self.true(filename.exists())
-        self.isin(mesg, mesgs)
-        self.notin(f'Schema {CORE_URL} already exists in local cache, skipping.', mesgs)
+        with self.getTestDir(copyfrom=s_getrefs.BASEDIR) as dirn:
+
+            filename = pathlib.Path(s_common.genpath(
+                dirn,
+                'raw.githubusercontent.com',
+                'oasis-open',
+                'cti-stix2-json-schemas',
+                'stix2.1',
+                'schemas',
+                'common',
+                'core.json'
+            ))
+
+            self.true(filename.exists())
+            filename.unlink()
+            self.false(filename.exists())
+
+            # Clear the cached validator funcs so the ref handlers in s_getrefs get called
+            s_config._JsValidators = {}
+
+            with self.getLoggerStream('synapse.utils.getrefs') as stream:
+                with mock.patch('synapse.utils.getrefs.BASEDIR', dirn):
+                    s_getrefs.main(args)
+
+            stream.seek(0)
+            mesgs = stream.read()
+            mesg = f'Downloading schema from {CORE_URL}.'
```

Comment:
Does this mean that we pulled the file from github during the test, instead of from the vcr recording?

Reply:
No, it was getting it from the cassette but the message still showed the original url

Reply:
hmm. I think you're right. I need to put the cassette back. I originally intended to get it from a local webserver but changed that and now it needs the cassette.